modules/install/pages/upgrade-cluster-online-full-capacity.adoc

```diff
@@ -0,0 +1,236 @@
+= Upgrade a Full-Capacity, Online Cluster
+
+:description: A cluster can be upgraded while still online; and maintained at full capacity.
+:tabs:
+
+[abstract]
+{description}
+
+[#online-upgrade-with-full-capacity]
+== Understanding Full-Capacity Upgrade
+
+The context and overall requirements for upgrading a live cluster by means of the introduction of one or more additional nodes are described in xref:install:upgrade-strategies.adoc[Upgrade-Procedure Selection]: this fully explains the node-by-node upgrade of the cluster, using _swap rebalance_ to minimize overhead.
+It also explains how, if additional nodes are available, the procedure can be completed with cluster performing at full capacity, during the upgrade.
+
+The precise steps for this procedure are detailed on this page, below.
+A full understanding of the information in xref:install:upgrade-strategies.adoc[Upgrade-Procedure Selection] should be acquired, before proceeding.
+The procedure assumes that:
+
+* The cluster to be upgraded must continue to serve data throughout the cluster-upgrade process.
+
+* A single, additional node is available; to act as a _swap_ node.
+The node _either_ has not previously run Couchbase Server, _or_ has run a version that can, with all its configuration files and other data, be overwritten: therefore, the latest version of Couchbase Server will be installed on the node.
+(Prior to installation, if necessary, remove any old configuration files from `/opt/couchbase`, using the command `rm -r /opt/couchbase`.)
+
+* During the cluster-upgrade process, _swap_ nodes and _upgraded_ nodes will be introduced to the cluster by the _node-addition_ procedure; while _failed over spare nodes_ and _nodes to be upgraded_ will be withdrawn from the cluster by means of the _node removal_ and _swap rebalance_ procedures.
++
+For information on node-addition, see xref:learn:clusters-and-availability/nodes.adoc#clusters[Clusters].
+For information on node-removal, see xref:learn:clusters-and-availability/removal.adoc[Removal].
+For information on _swap rebalance_, see xref:install:upgrade-strategies.adoc#swap-rebalance[Swap Rebalance].
+
+* Nodes will be upgraded _one at a time_.
+
+The overall procedure is described in seven stages, below.
```

Comment:
Not really, that's the problem. I completely agree it's a liability. But right now, asciidoc limits what we can do. We can only automate the numbering of paragraphs. I'd must sooner divide the paras into sections, rather than have nested numbered-paras (which are hard to read, and which asciidoc is buggy in its handling of). We should really be able to number sections, but can't yet. What I will do is open a ticket to get this looked at (we've actually talked about this, but haven't scheduled the contractor's time).